logtape/logger.ts

```diff
@@ -46,9 +47,7 @@ export interface Logger {
    * @param subcategory The subcategory.
    * @returns The child logger.
    */
-  getChild(
-    subcategory: string | readonly [string] | readonly [string, ...string[]],
-  ): Logger;
+  getChild(subcategory: Category): Logger;
```

Comment:
We should not allow empty arrays here.

Reply:
why?
I have a case where I generate category dynamically and it can be empty. In the case of this lib there is no real difference between returning current logger from getChild or returning a child as new instance. We just need to get the correct logger.

Reply:
In my fork I even made that if a category is empty string, empty array, null or undefined then it will return current logger instead of new child
Seems ok by me)

Reply:
I think it's misleading that the method name is `getChild()` but the object returned is clearly not a child logger, and I don't want the signature of the method to change.  🤔